python/GafferUI/CompoundEditor.py

```diff
@@ -355,6 +347,25 @@ def __removeCurrentTab( self, tabbedContainer ) :
 		# which are going to die.
 		GafferUI.EventLoop.addIdleCallback( self.__collect )
 
+	def __updateTabVisibility( self, tabbedContainer, tabsVisible ) :
+
+		tabbedContainer.setTabsVisible( tabsVisible )
+
+		# This is a shame-faced hack to make sure the timeline in the default layout can't be compressed
+		# or stretched vertically. Fixing this properly is quite involved, because we'd need to find a sensible
+		# generic way for TabbedContainer to set a min/max height based on it's children, and then a sensible
+		# generic rule for what SplitContainer should do in its __applySizePolicy() method.
+		if len( tabbedContainer ) == 1 and isinstance( tabbedContainer[0], GafferUI.Timeline ) :
+			if not tabsVisible :
+				# Fix height so timeline is not resizable
+				tabbedContainer._qtWidget().setFixedHeight( tabbedContainer[0]._qtWidget().sizeHint().height() )
+				tabbedContainer.parent()._qtWidget().setSizePolicy( QtWidgets.QSizePolicy.Ignored, QtWidgets.QSizePolicy.Fixed )
+			else :
+				# Undo the above
+				QWIDGETSIZE_MAX = 16777215 # Macro not exposed by Qt.py, but needed to remove fixed height
+				tabbedContainer._qtWidget().setFixedHeight( 16777215 )
```

Comment:
Were you meaning to set `QWIDGETSIZE_MAX` here rather than duplicate the int?

Reply:
Gah. I'm a muppet. Force-pushed a fix...